src/tree/utils/populateTree.ts

```diff
@@ -245,7 +246,22 @@ function prepareSchema(
   try {
     return resolveSchema(schema, path, options);
   } catch (ex) {
-    (node as TreeListParentNode).children = [];
-    return void (node as TreeListParentNode).children.push(createErrorTreeNode(node as TreeListParentNode, ex.message));
+    const treeNode: SchemaTreeListNode = {
+      id: generateId(),
+      name: '',
+      parent: node as TreeListParentNode,
+      children: [],
+    };
+
+    (node as TreeListParentNode).children.push(treeNode);
+    metadataStore.set(treeNode, {
+      schemaNode: processNode(schema || {}),
+      schema: schema || {},
+      path,
+    });
+
+    treeNode.children.push(createErrorTreeNode(treeNode, ex.message));
+
+    return null;
```

Comment:
You used to return `undefined`, but now it returns `null`. I guess this was not intended.

By the way this return statement is a **lot** nicer then the `return void doStuff()` we had before. 👍

Reply:
I guess this is one of those methods that return `T | null | undefined` where really it could get away with either `T | null` or - preferably - with `T | undefined`. If it is the case, let's narrow it down while we're here, it helps every future reader / user IMO.

Reply:
Somehow I like this `return void (call something)` thing, though never used it so far.

I used `null` as that's the return type of this function and it is still fine with the object checking in `populateTree`.  Previous `undefined` didn't match it. We use `JSONSchema4 | null` in few other places so I would keep it for now.

Reply:
Just curious why do you prefer it over the much more explicit:
```ts
doSomething();
return;
```
?

Anyway, I still don't like that something can return both `null` **or** `undefined` but I see that's how the whole infrastructure is built up here. This is question out of scope for this PR then, go ahead and merge it.